pycheribuild/config/compilation_targets.py

```diff
@@ -199,9 +200,11 @@ def riscv_arch_string(self):
         # Use the insane RISC-V arch string to enable CHERI
         if self.is_baremetal():
             # Baremetal/FreeRTOS only supports softfloat
-            arch_string = "rv64imac"
+            arch_string = "rv64imac" if self.target.cpu_architecture == CPUArchitecture.RISCV64 else \
+                          "rv32imac"
```

Comment:
```suggestion
            if self.target.cpu_architecture == CPUArchitecture.RISCV64:
                arch_string = "rv64imac"
            else:
                arch_string = "rv32imac"
```
looks better I think given the long condition (repeated for all the similar diffs below).